internal/controllers/clusterclass/clusterclass_controller.go

```
@@ -177,25 +195,70 @@ func (r *Reconciler) reconcile(ctx context.Context, clusterClass *clusterv1.Clus
+		Name: variable.Name,
+		// TODO: In a future iteration this should be false where definitions are equal.
+		DefintionsConflict: true,
+		Definitions: []clusterv1.ClusterClassStatusVariableDefinition{
+			{
+				From:     from,
+				Required: variable.Required,
+				Schema:   variable.Schema,
+			},
+		}}
+}
+
+func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clusterv1.ClusterClass) []error {
+	errs := []error{}
+
+	// Add inline variable definitions to the ClusterClass status.
+	clusterClass.Status.Variables = []clusterv1.ClusterClassStatusVariable{}
+	for _, variable := range clusterClass.Spec.Variables {
+		clusterClass.Status.Variables = append(clusterClass.Status.Variables, statusVariableFromClusterClassVariable(variable, clusterv1.VariableDefinitionFromInline))
+	}
+
+	// If RuntimeSDK is enabled call the DiscoverVariables hook for all associated Runtime Extensions and add the variables
+	// to the ClusterClass status.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		for _, patch := range clusterClass.Spec.Patches {
+			if patch.External == nil || patch.External.DiscoverVariablesExtension == nil {
+				continue
+			}
+			req := &runtimehooksv1.DiscoverVariablesRequest{}
+			req.Settings = patch.External.Settings
+
+			resp := &runtimehooksv1.DiscoverVariablesResponse{}
+			err := r.RuntimeClient.CallExtension(ctx, runtimehooksv1.DiscoverVariables, clusterClass, *patch.External.DiscoverVariablesExtension, req, resp)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "failed to call DiscoverVariables for patch %s", patch.Name))
+			}
+			if resp.Status != runtimehooksv1.ResponseStatusSuccess {
+				errs = append(errs, errors.Errorf("failed to discover variables for ClusterClass %s: %s", clusterClass.Name, resp.Message))
+			}
+			if resp.Variables != nil {
+				for _, variable := range resp.Variables {
+					// TODO: Variables should be validated and deduplicated base on their provided definition.
```

What about temporarily making it fail in case of duplicate definitions?
This would avoid silent failures down the line